src/Symfony/Bundle/SecurityBundle/DependencyInjection/SecurityExtension.php

```diff
@@ -424,6 +425,14 @@ private function createFirewall(ContainerBuilder $container, string $id, array $
         $container->register('security.user_checker.chain.'.$id, ChainUserChecker::class)
             ->addArgument(new TaggedIteratorArgument('security.user_checker.'.$id));
 
+        // Register Firewall-specific user checker for 'security.helper' login()
+        $securityUserCheckerLocator = $container->getDefinition('security.user_checker_locator');
+        $securityUserCheckerLocator
+            ->replaceArgument(0, array_merge($securityUserCheckerLocator->getArgument(0), [
+                'security.user_checker.'.$id => new ServiceClosureArgument(new Reference('security.user_checker.'.$id)),
```

Comment:
You should reuse the chain defined just before

```
security.user_checker.chain.'.$id
```

Reply:
Thx for your comment 
I fear not getting it, you mean i should add it in the locator or i should not use locator at all but inject this chaines instead and use it ?

Reply:
I'm not sure a locator is needed. There are already "[ChainChecker](https://github.com/symfony/symfony/blob/7.1/src/Symfony/Component/Security/Core/User/ChainUserChecker.php)" registered per firewall, that have all the checkers you reference there.

The problem is -imho-  that no Event is triggered during this execution (there is one for the standard behaviour (see [UserCheckerListener](https://github.com/symfony/symfony/blob/7.1/src/Symfony/Component/Security/Http/EventListener/UserCheckerListener.php)) login or SwitchUser) So the checkPreAuth is called only with the "main" UserChecker and not the one registered, as you already pointed.

I'm thinking instead of adding services there, it would maybe be prefereable to add a new Event ("manualLogin" -- with a better name haha)  and plug the checkers in the same way they are called in the authentication events.

Reply:
@smnandre thx for detailing this, I got it more
I know @chalasr and @wouterj are a lot of security subject so I would like to have more hints

my PR just want to fix the bug/security issue

Reply:
Thanks for the ping.

> I'm thinking instead of adding services there, it would maybe be prefereable to add a new Event

Indeed, relying on `UserCheckerListener` sounds better. Maybe make it listen to a `CheckUserEvent` in addition to `CheckPassportEvent`? 
As this is a bugfix, that new event class should be internal at first. On 7.x we'll be able to make it public, dispatch it in the regular authentication flow and deprecate listening on `CheckPassportEvent` in `UserCheckerListener`.

Reply:
I will try to find some time this week to address both of your reviews 
Thx